File: App/Resources/AlgorithmDetails/bubblebogosort/bubblebogosort.cs
using System;
public class BubbleBogoSort
{
  public static void Sort(int[] a)
  {
    int n = a.Length; if (n < 2) return;
    bool swapped = true;
    while (swapped)
    {
      swapped = false;
      for (int i = 0; i + 1 < n; i++) if (a[i] > a[i + 1])
      {
        int held = a[i]; a[i] = a[i + 1]; a[i + 1] = held; swapped = true;
      }
    }
  }
  public static void Main()
  {
    int[] array = { 0, 39, 21, 62, 91, 77, 14, 23 }; Sort(array);
    Console.WriteLine("[" + string.Join(", ", array) + "]");
  }
}

File: App/Resources/AlgorithmDetails/bubblebogosort/bubblebogosort.js
function sort(a) {
  const n = a.length;
  if (n < 2) return;
  let swapped = true;
  while (swapped) {
    swapped = false;
    for (let i = 0; i + 1 < n; i++) {
      if (a[i] > a[i + 1]) {
        [a[i], a[i + 1]] = [a[i + 1], a[i]];
        swapped = true;
      }
    }
  }
}
const array = [
  0, 39, 21, 62, 91, 77, 14, 23,
];
sort(array);
console.log("[" + array.join(", ") + "]");
